apps/challenges/views.py

```diff
@@ -2808,3 +2560,421 @@ def get_annotation_file_presigned_url(request, challenge_phase_pk):
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # Extract zip file
+    try:
+        zip_ref = extract_zip_file(
+            CHALLENGE_ZIP_DOWNLOAD_LOCATION,
+            "r",
+            join(BASE_LOCATION, unique_folder_name),
+        )
+    except zipfile.BadZipfile:
+        message = "The zip file contents cannot be extracted. Please check the format!"
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    error_messages, yaml_file_data, files = validate_challenge_config_util(
+        request,
+        challenge_host_team,
+        BASE_LOCATION,
+        unique_folder_name,
+        zip_ref,
+    )
+
+    if not len(error_messages):
+        if not challenge_queryset:
+            try:
+                with transaction.atomic():
+                    serializer = ZipChallengeSerializer(
+                        data=yaml_file_data,
+                        context={
+                            "request": request,
+                            "challenge_host_team": challenge_host_team,
+                            "image": files["challenge_image_file"],
+                            "evaluation_script": files[
+                                "challenge_evaluation_script_file"
+                            ],
+                            "github_repository": request.data[
+                                "GITHUB_REPOSITORY"
+                            ],
+                        },
+                    )
+                    if serializer.is_valid():
+                        serializer.save()
+                    challenge = serializer.instance
+                    queue_name = get_queue_name(challenge.title, challenge.pk)
+                    challenge.queue = queue_name
+                    challenge.save()
+
+                    # Create Leaderboard
+                    yaml_file_data_of_leaderboard = yaml_file_data[
+                        "leaderboard"
+                    ]
+                    leaderboard_ids = {}
+                    for data in yaml_file_data_of_leaderboard:
+                        serializer = LeaderboardSerializer(
+                            data=data, context={"config_id": data["id"]}
+                        )
+                        if serializer.is_valid():
+                            serializer.save()
+                        leaderboard_ids[
+                            str(data["id"])
+                        ] = serializer.instance.pk
+
+                    # Create Challenge Phase
+                    challenge_phase_ids = {}
+                    challenge_phases_data = yaml_file_data["challenge_phases"]
+                    for data, challenge_test_annotation_file in zip(
+                        challenge_phases_data,
+                        files["challenge_test_annotation_files"],
+                    ):
+                        data["slug"] = "{}-{}-{}".format(
```

Comment:
isn't there a util method to get slug?

Reply:
That's different. That just replaces empty spaces with '-', and is used only for challenge. I can use it here only to modify the codename.